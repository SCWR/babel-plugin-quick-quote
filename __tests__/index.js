/* global it expect:true */
let babel = require('@babel/core')
let plugin = require('../lib/index.js')
let {QUICK_QUOTE_KEY, QUICK_QUOTE_ASYNC_KEY} = require('../lib/tool.js')

function transform (source) {
  return babel.transform(source, { presets: [], plugins: [plugin] })
}

const example = [
  `
    const echarts = ${QUICK_QUOTE_KEY}([
      'path/demo/path',
      'line',
      'bar'
    ])
  `,
  `
    const echarts = ${QUICK_QUOTE_KEY}([
      'line',
      'bar'
    ]), test = 'test';
  `,
  `
    ${QUICK_QUOTE_KEY}([
      'line',
      'bar'
    ])
  `,
  `
    const initEcharts = ${QUICK_QUOTE_ASYNC_KEY}([
      'line',
      'bar'
    ])
  `,
  `
    ${QUICK_QUOTE_ASYNC_KEY}([
      'line',
      'bar'
    ])
  `,
]

example.forEach((source) => {
  it(`${source} should works to`, () => {
    const { code } = transform(source)
    expect(code).toMatchSnapshot()
  })
})
