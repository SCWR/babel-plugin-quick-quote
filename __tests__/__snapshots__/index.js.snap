// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`
    $_quickQuote([
      'line',
      'bar'
    ])
   should works to 1`] = `
"import \\"echarts/lib/echarts\\";
import \\"echarts/lib/chart/line\\";
import \\"echarts/lib/chart/bar\\";"
`;

exports[`
    $_quickQuoteAsync([
      'line',
      'bar'
    ])
   should works to 1`] = `
"let $_QuickQuote_echartsInstance = null;

function getEchartsInstance() {
  if (!$_QuickQuote_echartsInstance) {
    $_QuickQuote_echartsInstance = import(
    /* webpackChunkName: \\"echarts\\" */
    'echarts/lib/echarts').then(echarts => {
      return Promise.all([echarts, import(
      /* webpackChunkName: \\"echarts\\" */
      'echarts/lib/chart/line'), import(
      /* webpackChunkName: \\"echarts\\" */
      'echarts/lib/chart/bar')]).then(([echarts]) => echarts);
    });
  }

  return $_QuickQuote_echartsInstance;
}"
`;

exports[`
    const echarts = $_quickQuote([
      'line',
      'bar'
    ]), test = 'test';
   should works to 1`] = `
"import echarts from \\"echarts/lib/echarts\\";
import \\"echarts/lib/chart/line\\";
import \\"echarts/lib/chart/bar\\";
const test = 'test';"
`;

exports[`
    const echarts = $_quickQuote([
      'path/demo/path',
      'line',
      'bar'
    ])
   should works to 1`] = `
"import echarts from \\"echarts/lib/echarts\\";
import \\"path/demo/path\\";
import \\"echarts/lib/chart/line\\";
import \\"echarts/lib/chart/bar\\";"
`;

exports[`
    const initEcharts = $_quickQuoteAsync([
      'line',
      'bar'
    ])
   should works to 1`] = `
"let $_QuickQuote_echartsInstance = null;

function initEcharts() {
  if (!$_QuickQuote_echartsInstance) {
    $_QuickQuote_echartsInstance = import(
    /* webpackChunkName: \\"echarts\\" */
    'echarts/lib/echarts').then(echarts => {
      return Promise.all([echarts, import(
      /* webpackChunkName: \\"echarts\\" */
      'echarts/lib/chart/line'), import(
      /* webpackChunkName: \\"echarts\\" */
      'echarts/lib/chart/bar')]).then(([echarts]) => echarts);
    });
  }

  return $_QuickQuote_echartsInstance;
}"
`;
